extensions/amp-story/1.0/test/test-amp-story-tooltip.js

```diff
@@ -0,0 +1,189 @@
+  let tooltip;
+  let win;
+  let parentEl;
+  let storeService;
+  let fakePage;
+  let clickableEl;
+
+  beforeEach(() => {
+    win = env.win;
+    storeService = new AmpStoryStoreService(win);
+    registerServiceBuilder(win, 'story-store', () => storeService);
+    clickableEl = win.document.createElement('a');
+
+    // Making sure resource tasks run synchronously.
+    sandbox.stub(Services, 'resourcesForDoc').returns({
+      mutateElement: (element, callback) => {
+        callback();
+        return Promise.resolve();
+      },
+      measureMutateElement: (measure, mutate) => {
+        return Promise.resolve().then(measure).then(mutate);
+      },
+    });
+
+    parentEl = win.document.createElement('div');
+    win.document.body.appendChild(parentEl);
+
+    fakePage = win.document.createElement('amp-story-page');
+    addAttributesToElement(fakePage, {'active': ''});
+
+    parentEl.appendChild(fakePage);
+
+    tooltip = new AmpStoryTooltip(win, parentEl);
+  });
+
+  it('should build the tooltip', () => {
+    tooltip.build_();
+    expect(tooltip.isBuilt_).to.be.true;
+    expect(tooltip.tooltipOverlayEl_).to.exist;
+  });
+
+  it('should append the tooltip to the parentEl when clicking a clickable ' +
+    'element', () => {
+    fakePage.appendChild(clickableEl);
+
+    storeService.dispatch(Action.TOGGLE_TOOLTIP, clickableEl);
+
+    // Children in parentEl: fakePage and tooltip overlay.
+    expect(parentEl.childElementCount).to.equal(2);
+  });
+
+  it('should show the tooltip on store property update', () => {
+    fakePage.appendChild(clickableEl);
+
+    storeService.dispatch(Action.TOGGLE_TOOLTIP, clickableEl);
+
+    expect(tooltip.tooltipOverlayEl_).to.not.have.class('i-amphtml-hidden');
+  });
+
+  it('should hide the tooltip on store property update', () => {
+    fakePage.appendChild(clickableEl);
+
+    storeService.dispatch(Action.TOGGLE_TOOLTIP, clickableEl);
+    storeService.dispatch(Action.TOGGLE_TOOLTIP, null);
+
+    expect(tooltip.tooltipOverlayEl_).to.have.class('i-amphtml-hidden');
+  });
+
+  it('should hide the tooltip when switching page', () => {
+    fakePage.appendChild(clickableEl);
+    storeService.dispatch(Action.TOGGLE_TOOLTIP, clickableEl);
+
+    storeService.dispatch(Action.CHANGE_PAGE, {id: 'newPageId'});
+
+    expect(tooltip.tooltipOverlayEl_).to.have.class('i-amphtml-hidden');
+  });
+
+  it('should hide the tooltip when clicking outside of it', () => {
+    fakePage.appendChild(clickableEl);
+    storeService.dispatch(Action.TOGGLE_TOOLTIP, clickableEl);
+
+    tooltip.tooltipOverlayEl_.dispatchEvent(new Event('click'));
+
+    expect(tooltip.tooltipOverlayEl_).to.have.class('i-amphtml-hidden');
+  });
+
+  it('should navigate when tooltip is open and user clicks on arrow', () => {
+    fakePage.appendChild(clickableEl);
+    storeService.dispatch(Action.TOGGLE_TOOLTIP, clickableEl);
+    const fakeNavCall = sandbox.stub(tooltip, 'onNavigationalClick_');
```

Comment:
Could you please listen for the navigation event to be dispatched instead please? We should not stub private methods, they're part of the implementation.